app/waf/src/controllers/schema.ts

```diff
@@ -0,0 +1,284 @@
+import {ParameterObject, ParameterLocation} from '@loopback/openapi-v3-types';
+import {MetadataInspector} from '@loopback/metadata';
+import {PropertyDefinition, PropertyMap} from '@loopback/repository';
+import {Adc} from '../models';
+
+function buildProperties(
+  metadata: PropertyMap | undefined,
+  propNames: string[],
+): {[key: string]: Object} {
+  let props: {[key: string]: Object} = {};
+
+  if (metadata) {
+    for (let propName of propNames) {
+      props[propName] = {
+        type: metadata[propName].type,
+      };
+    }
+  }
+
+  return props;
+}
+
+const adcProperties = MetadataInspector.getAllPropertyMetadata<
```

Comment:
I'm not familiar with MetadataInspector. 

In fact, there may be possible thinking, may be helpful to you to make the implementation easier. 

In our model definition, it supports define our own meta properties, like this: 
```
// ... from application.model.ts
  @property({
    type: 'array',
    itemType: 'string',
    required: false,
    default: [],
    inResp: true,
    inReq: false
  })
  services: string[];
```

```
console.log(JSON.stringify(Application.definition.properties));
// output
... "services":{"type":"array","itemType":"string","required":false,"default":[],"inResp":true,"inReq":false}, ...
```

I'm very concern about this point. because: 
It would be a big workload if don't design carefully on the documentation process...
Can we make it possible to cut down ANY duplicate code pieces as possible as we can?

Reply:
We can attempt to define our own decorator to do that, but that is not containable in this pull request. Suggest to open another branch to do that.